src/containers/Rewards.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import {connect} from 'react-redux'
+import AddRewardForm from '../components/AddRewardForm';
+import {addContractData } from '../redux/actions/addReward';
+
+class AddReward extends Component{
+
+    state = {
+        amount: 0,
+        withdrawable: 0,
+    };
+
+    async componentDidMount(){
+        const {addContractDataFunc, contract: {data}} = this.props;
+        if (!data){
+            await addContractDataFunc()
+        }
+    };
+
+    handleChange = (e) => {
+        const {name, value} = e.target;
+        this.setState({[name]: value});
+    } 
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        const { amount, withdrawable} = this.state;
+        const {contract:{ data:{ owner, festaking}}} = this.props;
+
+        const totalReward =  await festaking.methods.totalReward().call();
+        console.log('total reward', totalReward);
+
+        const addReward = await festaking.methods.addReward(amount,withdrawable)
```

Comment:
I would suggest you create a proxy wrapper class to interact with the contract. So your code will look like this:

1. In the component did mount initialize the contract and listen to the changes:
const contract = FestakingProxy.create()
contract.onTransactionUpdate( (tx, events) => ... ) // Process updates of the transaction such as confirmations, failures, etc.

Then when required just call
const result = await contract.addReward(amount, withdrawable) // This will set the gas and send and starts monitoring for updates in the transaction
...

Make sure that UI reacts to updates in the transaction. Remember, in actual testnet or mainnet, after contract is submitted it will not be mined immediately